### PR TITLE
Support NodeJS 16 building and testing on RHEL9 host

### PR DESCRIPTION
**Pull request by phracek**
This pull request adds support for building and testing NodeJS 16 container on RHEL9 host.

The pull request also updates README.md files with RHEL9 and Quay.io references.

Difference between `16/Dockerfile.c9s` and `16/Dockerfile.rhel9`:
```bash
$ diff 16/Dockerfile.c9s 16/Dockerfile.rhel9
1c1
< FROM quay.io/sclorg/s2i-core-c9s:c9s
---
> FROM ubi9/s2i-core:1
24c24
<     CNB_STACK_ID=com.redhat.stacks.c9s-nodejs-16 \
---
>     CNB_STACK_ID=com.redhat.stacks.ubi9-nodejs-16 \
29c29
< DESCRIPTION="Node.js $NODEJS_VERSION available as container is a base platform for \
---
>     DESCRIPTION="Node.js $NODEJS_VERSION available as container is a base platform for \
44c44
<       io.buildpacks.stack.id="com.redhat.stacks.c9s-nodejs-16" \
---
>       io.buildpacks.stack.id="com.redhat.stacks.ubi9-nodejs-16" \
49c49
<       name="sclorg/$NAME-$NODEJS_VERSION" \
---
>       name="ubi9/$NAME-$NODEJS_VERSION" \
54c54
<       usage="s2i build <SOURCE-REPOSITORY> quay.io/sclorg/$NAME-$NODEJS_VERSION-c9s:latest <APP-NAME>"
---
>       usage="s2i build <SOURCE-REPOSITORY> ubi9/$NAME-$NODEJS_VERSION:latest <APP-NAME>"
56d55
< # Package libatomic_ops was removed
```

Difference between `16/Dockerfile.rhel8` and `16/Dockerfile.rhel9`:
```bash
$ diff 16/Dockerfile.rhel8 16/Dockerfile.rhel9
1c1
< FROM ubi8/s2i-core:1
---
> FROM ubi9/s2i-core:1
24c24
<     CNB_STACK_ID=com.redhat.stacks.ubi8-nodejs-16 \
---
>     CNB_STACK_ID=com.redhat.stacks.ubi9-nodejs-16 \
44c44
<       io.buildpacks.stack.id="com.redhat.stacks.ubi8-nodejs-16" \
---
>       io.buildpacks.stack.id="com.redhat.stacks.ubi9-nodejs-16" \
49c49
<       name="ubi8/$NAME-$NODEJS_VERSION" \
---
>       name="ubi9/$NAME-$NODEJS_VERSION" \
54c54
<       usage="s2i build <SOURCE-REPOSITORY> ubi8/$NAME-$NODEJS_VERSION:latest <APP-NAME>"
---
>       usage="s2i build <SOURCE-REPOSITORY> ubi9/$NAME-$NODEJS_VERSION:latest <APP-NAME>"
56,57c56
< RUN yum -y module enable nodejs:$NODEJS_VERSION && \
<     MODULE_DEPS="make gcc gcc-c++ libatomic_ops git openssl-devel" && \
---
> RUN MODULE_DEPS="make gcc gcc-c++ git openssl-devel" && \
60d58
<     ln -s /usr/libexec/platform-python /usr/bin/python3 && \
```

### CODE REVIEW

**Comment by phracek** [comment]
[test-all]

**Comment by phracek** [comment]
Container test results are:
```bash
 [PASSED] for 'app' test_run_app_application
 [PASSED] for 'app' test_s2i_usage
 [PASSED] for 'app' test_scl_usage
 [PASSED] for 'app' test_connection
 [PASSED] for 'app' test_docker_run_usage
 [PASSED] for 'app' test_scl_variables_in_dockerfile
 [PASSED] for 'app' test_npm_functionality
 [PASSED] for 'app' test_check_build_using_dockerfile
 [PASSED] for 'app' test_nodemon_removed
 [PASSED] for 'app' test_npm_cache_cleared
 [PASSED] for 'app' test_npm_tmp_cleared
 [PASSED] for 'app' kill_test_application
 [PASSED] for 'app' test_dev_mode_true_development
 [PASSED] for 'app' test_dev_mode_false_production
 [PASSED] for 'node_env_development' test_run_app_application
 [PASSED] for 'node_env_development' test_connection
 [PASSED] for 'node_env_development' test_nodemon_present
 [PASSED] for 'node_env_development' test_npm_cache_exists
 [PASSED] for 'node_env_development' kill_test_application
 [PASSED] for 'node_env_development' test_dev_mode_true_development
 [PASSED] for 'node_env_development' test_dev_mode_false_development
 [PASSED] for 'dev_mode' test_run_app_application
 [PASSED] for 'dev_mode' test_connection
 [PASSED] for 'dev_mode' test_nodemon_present
 [PASSED] for 'dev_mode' test_npm_cache_exists
 [PASSED] for 'dev_mode' kill_test_application
 [PASSED] for 'dev_mode' test_dev_mode_true_development
 [PASSED] for 'dev_mode' test_dev_mode_false_production
 [PASSED] for 'hw' test_safe_logging
 [PASSED] for 'hw' test_run_hw_application
 [PASSED] for 'hw' test_incremental_build
 [PASSED] for 'hw' test_build_express_webapp
 [PASSED] for 'clients' express
 [PASSED] for 'clients' pino
 [PASSED] for 'clients' prom-client
 [PASSED] for 'clients' opossum
 [PASSED] for 'clients' kube-service-bindings
 [PASSED] for 'binary' test_run_binary_application

Tests for ubi9/nodejs-16:1 succeeded.
```

OpenShift test results are:
```bash
-----------------------------------------------
Running test test_nodejs_imagestream (starting at 2022-06-20 13:34:32+02:00) ...
-----------------------------------------------
Imagestream testing not supported for rhel9 environment.
 [PASSED] for 'openshift-remote-cluster' test_nodejs_s2i_container (00:00:59)
 [PASSED] for 'openshift-remote-cluster' test_nodejs_s2i_app_ex (00:01:00)
 [PASSED] for 'openshift-remote-cluster' test_nodejs_s2i_templates (00:03:29)
 [PASSED] for 'openshift-remote-cluster' test_nodejs_imagestream (00:00:00)

OpenShift tests for ubi9/nodejs-16:1 succeeded.
```

File: 16/README.md [hunk 1]
```diff
@@ -5,7 +5,7 @@ This container image includes Node.JS 16 as a [S2I](https://github.com/openshift
 Users can choose between RHEL, CentOS and Fedora based images.
 The RHEL images are available in the [Red Hat Container Catalog](https://access.redhat.com/containers/),
 the CentOS images are available on [Quay.io](https://quay.io/organization/centos7),
-and the Fedora images are available in [Fedora Registry](https://registry.fedoraproject.org/).
+and the Fedora images are available in [Fedora Registry](https://quay.io/organization/fedora).
```
Comment (ljavorsk) [inline 16/README.md:8]: The `Fedora Registry` should be `Quay.io Registry`

**Comment by hhorak** [comment]
LGTM, no issue spotted

**Comment by ljavorsk** [comment]
Maybe add some comment to the Dockerfile why the ``libatomic_ops`` are missing from ``MODULE_DEPS``.

Otherwise LGTM :)

**Comment by phracek** [comment]
Because liatomic_ops does not exist in RHEL9 repositories. But I will change README.md. Thanks

**Comment by phracek** [comment]
[test-all]

**Comment by phracek** [comment]
Fedora tests have failed on `pino`:

```bash
10:03:53                 out:  [PASSED] for 'hw' test_incremental_build
10:03:53                 out:  [PASSED] for 'hw' test_build_express_webapp
10:03:53                 out:  [PASSED] for 'clients' express
10:03:53                 out:  [FAILED] for 'clients' pino
10:03:53                 out:  [PASSED] for 'clients' prom-client
10:03:53                 out:  [PASSED] for 'clients' opossum
10:03:53                 out:  [PASSED] for 'clients' kube-service-bindings
10:03:53                 out:  [PASSED] for 'binary' test_run_binary_application
```

**Pull request merged**